packages/ethers-provider/src/store/index.ts

```diff
@@ -0,0 +1,84 @@
+import {
+  BehaviorSubject,
+  distinctUntilKeyChanged,
+  filter,
+  map,
+  Observable,
+  Subject,
+} from 'rxjs';
+
+import { APP_INITIAL_STATE } from '../constants';
+import { Action, AppState, Chain } from '../types';
+import { ADD_CHAINS, RESET_STORE, UPDATE_CHAIN } from './constants';
+
+const _store = new BehaviorSubject<AppState>(APP_INITIAL_STATE);
+const _stateUpdates = new Subject<AppState>();
+
+_stateUpdates.subscribe(_store);
+
+function reducer(state: AppState, action: Action): AppState {
+  const { type, payload } = action;
+
+  switch (type) {
+    case ADD_CHAINS:
+      return {
+        ...state,
+        chains: [...state.chains, ...(payload as Chain[])],
+      };
+
+    case UPDATE_CHAIN:
+      const chains = state.chains;
+      const current = chains.findIndex(({ id }) => id === payload.id);
+
+      if (current !== -1) {
+        chains[current] = payload as Chain;
+      } else {
+        chains.push(payload as Chain);
+      }
+
+      return {
+        ...state,
+        chains,
+      };
+
+    case RESET_STORE:
+      return APP_INITIAL_STATE;
+
+    default:
+      throw new Error(`Unknown type: ${type} in appStore reducer`);
+  }
+}
+
+export const dispatch = (action: Action) => {
+  const state = _store.getValue();
+  _stateUpdates.next(reducer(state, action));
+};
+
+function select(): Observable<AppState>;
+function select<T extends keyof AppState>(stateKey: T): Observable<AppState[T]>;
+function select<T extends keyof AppState>(
+  stateKey?: keyof AppState,
+): Observable<AppState[T]> | Observable<AppState> {
+  if (!stateKey) return _stateUpdates.asObservable();
```

Comment:
```suggestion
  if (!stateKey) {
     return _stateUpdates.asObservable();
  }
```
I know that not all of the code guidelines we have are applicable to this package but try to use those you can.